unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,176 @@
+        register_fixture.register()
+        register_fixture.register.assert_called_once()
+
+    def test_function_unregister_call_success(self, unregister_fixture):
+        unregister_fixture.unregister()
+        unregister_fixture.unregister.assert_called_once()
+
+    def test_function_on_register_call_success(self, on_register_fixture):
+        on_register_fixture._on_register()
+        on_register_fixture._on_register.assert_called_once()
+
+    def test_function__check_transaction_id_cached_out_call_success(self,
+                                                                    check_transaction_id_cached_out_call_success_fixture):
+        check_transaction_id_cached_out_call_success_fixture._check_transaction_id_cached_out()
+        check_transaction_id_cached_out_call_success_fixture._check_transaction_id_cached_out.assert_called_once()
+
+    def test_check_buffer_message_matching_command_and_id_throws_exception(self):
+        message = {}
+        with pytest.raises(RedisAPIException,
+                           match='The answer message does not contain a valid '
+                                 '\'transaction_id\' member.') as ex:
+            redis_client_base = RedisClientBase(area_id=area_id, autoregister=False)
+            redis_client_base._check_buffer_message_matching_command_and_id(message)
+
+    def test_register_blocking_true_throws_exception(self):
+        with pytest.raises(RedisAPIException,
+                           match='API registration process timed out. Server will continue processing '
+                                 'your request on the background and will notify you as soon as the '
+                                 'registration has been completed.') as ex:
+            redis_client_base = RedisClientBase(area_id=area_id, autoregister=True)
+            redis_client_base.register(is_blocking=True)
+
+    def test_register_blocking_false_published_success(self):
+        # arrange
+        redis_client_base = RedisClientBase(area_id=area_id, autoregister=False)
+        redis_client_base.register(is_blocking=False)
+        assert redis_client_base._blocking_command_responses["register"]["name"] == area_id
+
+    def test_register_blocking_false_throw_exception(self, redis_client_auto_register):
+        with pytest.raises(RedisAPIException,
+                           match='API is already registered to the market.') as ex:
+            self.redis_client_base = redis_client_auto_register
+            self.redis_client_base.is_active = True
+            self.redis_client_base.register(is_blocking=False)
+
+    def test_unregister_blocking_isactive_false_throws_exception(self, redis_client_auto_register):
+        with pytest.raises(RedisAPIException,
+                           match='API is already unregistered from the market.') as ex:
+            self.obj = redis_client_auto_register
+            self.obj.is_active = False
+            self.obj.unregister(is_blocking=True)
```

Comment:
here you are overwriting the self.obj with the fixture. No need, you can just use the fixture directly
```suggestion
            redis_client_auto_register.is_active = False
            redis_client_auto_register.unregister(is_blocking=True)
```

Reply:
The same applies to the following test functions.